Lib/mimetypes.py

```diff
@@ -94,8 +94,8 @@ def add_type(self, type, ext, strict=True):
         if ext not in exts:
             exts.append(ext)
 
-    def guess_type(self, url, strict=True):
-        """Guess the type of a file based on its URL.
+    def guess_type(self, url_or_path, strict=True):
+        """Guess the type of a file which is either a URL or an os.PathLike object.
```

Comment:
I would say "path-like object" rather than "os.PathLike object", since os.PathLike isn't something that users instantiate explicitly.
Also, I think the argument name can be kept as `url` rather than `url_or_path`. People may be passing the argument by name.